snapcraft/plugins/_python/_pip.py

```diff
@@ -99,9 +98,12 @@ def __init__(self, *, python_major_version, part_dir, install_dir,
             self._python_major_version, stage_dir=self._stage_dir,
             install_dir=self._install_dir)
 
-        self._setup()
```

Comment:
Good call, this makes testing easier :-)

Reply:
Yeah. I liked having an init that made sure the class was immediately usable, but then it was impossible to determine whether or not things were setup without... setting them up :stuck_out_tongue: .